### PR TITLE
sessionctx: support session variable tidb_read_staleness 

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What is changed and how it works?
Support session variable `tidb_read_staleness` which let the following query statement in stale read in current session.

### Check List 

Tests 

- [x] Unit test

### Release note 

```release-note
support session variable tidb_read_staleness
```

### CODE REVIEW

File: executor/stale_txn_test.go [hunk 1]
```diff
@@ -1106,3 +1106,37 @@ func (s *testStaleTxnSuite) TestStmtCtxStaleFlag(c *C) {
...
+	safePointComment := "All versions after safe point can be accessed. (DO NOT EDIT)"
+	updateSafePoint := fmt.Sprintf(`INSERT INTO mysql.tidb VALUES ('%[1]s', '%[2]s', '%[3]s')
+	ON DUPLICATE KEY
+	UPDATE variable_value = '%[2]s', comment = '%[3]s'`, safePointName, safePointValue, safePointComment)
+	tk.MustExec(updateSafePoint)
+	tk.MustExec("use test")
+	tk.MustExec("create table t10 (id int);")
+	time.Sleep(3 * time.Second)
+	now := time.Now()
+	tk.MustExec(`set @@tidb_read_staleness="-2s"`)
```
Comment (JmPotato) [inline executor/stale_txn_test.go:1124]: What about adding a test case about invalid input such as:

```go
set @@tidb_read_staleness="invalid_time"
```

File: executor/stale_txn_test.go [hunk 2]
```diff
@@ -1106,3 +1106,37 @@ func (s *testStaleTxnSuite) TestStmtCtxStaleFlag(c *C) {
...
+	c.Assert(failpoint.Enable("github.com/pingcap/tidb/executor/assertStaleTSO", fmt.Sprintf(`return(%d)`, now.Unix()-2)), IsNil)
+	tk.MustExec("select * from t10;")
+	c.Assert(failpoint.Disable("github.com/pingcap/tidb/executor/assertStaleTSO"), IsNil)
+	c.Assert(failpoint.Disable("github.com/pingcap/tidb/expression/injectNow"), IsNil)
+	// begin transaction won't be affected by read staleness
+	tk.MustExec("begin")
+	tk.MustExec("insert into t10(id) values (1);")
+	tk.MustExec("commit")
+	tk.MustExec("insert into t10(id) values (2);")
+	// query will still use staleness read
```
Comment (JmPotato) [inline executor/stale_txn_test.go:1136]: What about adding a case that we use `as of timestamp...` after setting `@@tidb_read_staleness`?

File: executor/stale_txn_test.go (hunk 1 above)
Comment (Yisaer) [reply executor/stale_txn_test.go]: updated.

File: executor/stale_txn_test.go (hunk 2 above)
Comment (Yisaer) [reply executor/stale_txn_test.go:1137]: The compatibility between different cases of stale read will be pushed in next request.

**Review by xhebox — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**